[Safari] [Liquid Glass] Turn Off The Lights extension has no color fill behind navigation bar

whsieh opened the pull request:
#### 6c4ef1d757f2a0e7f47d50684ddcbe818b252fa6
<pre>
[Safari] [Liquid Glass] Turn Off The Lights extension has no color fill behind navigation bar
<a href="https://bugs.webkit.org/show_bug.cgi?id=294262">https://bugs.webkit.org/show_bug.cgi?id=294262</a>
<a href="https://rdar.apple.com/152865474">rdar://152865474</a>

Reviewed by Abrar Rahman Protyasha.

Dial back some of the mitigations implemented in 295724@main that avoid detecting large viewport-
constrained containers for color extension. In the case of a dimming layer that covers the entire
viewport, failing to extend edge colors causes Safari UI to look broken, as there&apos;s either a gap or
different color around the dimming layer.

Keep both cases working by not rejecting large viewport-constrained elements in the case where they
look like they&apos;re intended to dim the webpage.

* LayoutTests/fast/page-color-sampling/color-sampling-ignores-large-sticky-container-expected.txt: Added.
* LayoutTests/fast/page-color-sampling/color-sampling-ignores-large-sticky-container.html: Added.

Add a new layout test to verify that a large, scrollable sticky container that contains content
doesn&apos;t trigger color extension.

* LayoutTests/fast/page-color-sampling/color-sampling-ignores-modal-popup-expected.txt: Removed.
* LayoutTests/fast/page-color-sampling/color-sampling-modal-popup-expected.txt: Added.
* LayoutTests/fast/page-color-sampling/color-sampling-modal-popup.html: Renamed from LayoutTests/fast/page-color-sampling/color-sampling-ignores-modal-popup.html.

Rename this test from `color-sampling-ignores-modal-popup.html` to `color-sampling-modal-popup.html`
and adjust it so that it expects colors to change as the modal container is presented.

* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::fixedContainerEdges const):

See above.

Canonical link: <a href="https://commits.webkit.org/296056@main">https://commits.webkit.org/296056@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bb8cdd14f456fe578014cc0f4770e9318f59faa5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/107179 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/26860 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/17260 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/112391 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/57716 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/109143 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/27575 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/35364 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81364 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110108 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/21871 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96625 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61758 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/21296 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/14781 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57157 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/91225 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14806 "Passed tests") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/115489 "Build was cancelled. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (cancelled)") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/34285 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25260 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90405 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/34657 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92881 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90139 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22990 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35078 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12839 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/29978 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34168 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/39643 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/33915 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/37267 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/35571 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (approved):
r+, small editorial comment

> in the case where **they** look like they're intended

whsieh committed:
[Safari] [Liquid Glass] Turn Off The Lights extension has no color fill behind navigation bar
https://bugs.webkit.org/show_bug.cgi?id=294262
rdar://152865474

Reviewed by Abrar Rahman Protyasha.

Dial back some of the mitigations implemented in 295724@main that avoid detecting large viewport-
constrained containers for color extension. In the case of a dimming layer that covers the entire
viewport, failing to extend edge colors causes Safari UI to look broken, as there's either a gap or
different color around the dimming layer.

Keep both cases working by not rejecting large viewport-constrained elements in the case where they
look like they're intended to dim the webpage.

* LayoutTests/fast/page-color-sampling/color-sampling-ignores-large-sticky-container-expected.txt: Added.
* LayoutTests/fast/page-color-sampling/color-sampling-ignores-large-sticky-container.html: Added.

Add a new layout test to verify that a large, scrollable sticky container that contains content
doesn't trigger color extension.

* LayoutTests/fast/page-color-sampling/color-sampling-ignores-modal-popup-expected.txt: Removed.
* LayoutTests/fast/page-color-sampling/color-sampling-modal-popup-expected.txt: Added.
* LayoutTests/fast/page-color-sampling/color-sampling-modal-popup.html: Renamed from LayoutTests/fast/page-color-sampling/color-sampling-ignores-modal-popup.html.

Rename this test from `color-sampling-ignores-modal-popup.html` to `color-sampling-modal-popup.html`
and adjust it so that it expects colors to change as the modal container is presented.

* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::fixedContainerEdges const):

See above.

Canonical link: https://commits.webkit.org/296056@main
